diff --git a/node_modules/react-native-modal/dist/modal.js b/node_modules/react-native-modal/dist/modal.js
index 80f4e75..a5974aa 100644
--- a/node_modules/react-native-modal/dist/modal.js
+++ b/node_modules/react-native-modal/dist/modal.js
@@ -65,6 +65,7 @@ export class ReactNativeModal extends React.Component {
         this.currentSwipingDirection = null;
         this.panResponder = null;
         this.didUpdateDimensionsEmitter = null;
+        this.backHandlerEventSubscription;
         this.interactionHandle = null;
         this.getDeviceHeight = () => this.props.deviceHeight || this.state.deviceHeight;
         this.getDeviceWidth = () => this.props.deviceWidth || this.state.deviceWidth;
@@ -453,10 +454,12 @@ export class ReactNativeModal extends React.Component {
         if (this.state.isVisible) {
             this.open();
         }
-        BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
+        this.backHandlerEventSubscription = BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
     }
     componentWillUnmount() {
-        BackHandler.removeEventListener('hardwareBackPress', this.onBackButtonPress);
+        if (this.backHandlerEventSubscription) {
+            this.backHandlerEventSubscription.remove();
+        }
         if (this.didUpdateDimensionsEmitter) {
             this.didUpdateDimensionsEmitter.remove();
         }
